ConfigSpace/read_and_write/pcs_new.py

```diff
@@ -414,11 +414,36 @@ def read(pcs_string, debug=False):
             else:
                 # So far, only equals is supported by SMAC
                 if tmp_list[1] == '=':
-                    # TODO maybe add a check if the hyperparameter is
-                    # actually in the configuration space
+                    hp = configuration_space.get_hyperparameter(tmp_list[0])
+                    if isinstance(hp, NumericalHyperparameter):
+                        if isinstance(hp, IntegerHyperparameter):
+                            forbidden_value = int(tmp_list[2])
+                        elif isinstance(hp, FloatHyperparameter):
+                            forbidden_value = float(tmp_list[2])
+                        if forbidden_value < hp.lower or forbidden_value > hp.upper:
+                            # TODO alternatively we could raise an warning and ignore this
+                            raise ValueError(f'forbidden_value is set out of the bound, it needs to'
+                                             f' be set between [{hp.lower}, {hp.upper}]'
+                                             f' but its value is {forbidden_value}')
+                    elif isinstance(hp, (CategoricalHyperparameter, OrdinalHyperparameter)):
+                        hp_values = hp.choices if isinstance(hp, CategoricalHyperparameter)\
+                            else hp.sequence
+                        forbidden_value_in_hp_values = False
+                        for hp_value in hp_values:
+                            if str(hp_value) == tmp_list[2]:
```

Comment:
When I implemented this code, I was thinking that hp_value could be a float or int. But since pcs_new reader only transform choices/ sequences to string, this line is redundant, I will fix that